### ISSUE TITLE
How to integrate into a fresh HA 2024.2 installation?

### ISSUE THREAD

**Issue by TomDue** [RalfWinter/lambda-heatpump-modbus-tcp-HA#9]
Hi, i'm new into HA and don't know how to exactly integrate your YAML file into the HA configuration. I found the "configuration.yaml" but it's almost empty. How do I exactly integrate the Lambda Heatpump?
Thanks for helping a newbie.

**Comment by RalfWinter** [comment]
Save the "HA-Lambda-WP_configuration.yaml" into the directory where your configuration.yaml resides and add following line to your configuration.yaml:
template: !include HA-Lambda-WP_configuration.yaml

**Comment by TomDue** [comment]
Hi Ralf
 
Thanks very much!
I can load it but I get this:
 
 Logger: homeassistant.config
Source: config.py:597
First occurred: 21:10:56 (1 occurrences)
Last logged: 21:10:56
 
Invalid config for 'template' at Lambda-WP.yaml, line 4: 'modbus' is an invalid option for 'template', check: modbus Invalid config for 'template' at Lambda-WP.yaml, line 558: 'platform' is an invalid option for 'template', check: sensor->0->platform Invalid config for 'template' at Lambda-WP.yaml, line 558: required key 'state' not provided Invalid config for 'template' at Lambda-WP.yaml, line 559: 'sensors' is an invalid option for 'template', check: sensor->0->sensors
 
I changed the your configuration to fit my Lambda IP and commented out the modules I don't have. Attached is my configuration. Maybe you see what I did wrong immediately?
 
Your help is appreciated a lot!
Thanks
Tom

**Comment by TomDue** [comment]
Hi Ralf
 
It works 
I get data.
 
Thanks a lot!
 
Regards
Tom

**Comment by RalfWinter** [comment]
Thanks for letting us know.

**Comment by bs0001** [comment]
My Lambda EU08L will be connected in the next 2 weeks.
Therefore I wan't to prepare my Home Assistant to finalize the settings when the heatpump will get an IP adress and is reachable in my local network.

Therefore I created in HA a folder "packages" where I uploaded the HA-Lambda-WP_configuration.yaml in it.
In my configuration.yaml file I do the following entry (it's my first package in HA!):

homeassistant:
  packages: !include_dir_named packages

When I check my configuration I get the following error:
Konfigurationsfehler
Error loading /config/configuration.yaml: mapping values are not allowed here
  in "/config/packages/HA-Lambda-WP_configuration.yaml", line 218, column 74

Whar is the reason and how can I solve it?

**Comment by RalfWinter** [comment]
Congratulation for your Lambda HP!
You need to append your configuration.yaml with the yaml I created for the Lambda heatpump.
HA does not support the inclusion of other yaml files.

**Comment by bs0001** [comment]
How should the configuration.yaml look like exactly woth the lambda entry?

**Comment by RalfWinter** [comment]
I will post mine later, when I have access to my installation.

**Comment by RalfWinter** [comment]
I inserted the code right after the first 11 lines: 
```
# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

# Text to speech
tts:
  - platform: google_translate

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

**# Lambda WP Modbus/TCP Integration**
```

**Comment by bs0001** [comment]
Hi Ralf,

thanks for your info!
My mistake was to download the file on NAS and then upload to HA.
This haven't worked.

I opened the HA-Lambda-WP_configuration.yaml directly on github and copied everything to notepad++.
Then I renamed EU13L to EU08L.

I put the whole bunch of renamed content in subdirectory HA --> packages.
With "-" and upper case the configuration was not accepted. I renamed your file to "ha_lambda_wp_configuration.yaml". Probably you should rename our file on github also!

With my entry in configuration.yaml all packages in the subdirectory will be proved and activated after restart:
homeassistant:
  packages: !include_dir_named packages


Anyway, I'm still waiting that my EU08L will be finally connected with all sensors and initally connected to my home. This hasen't been done yet.

After verifying the HA configuration I get the following messages:
Invalid config for 'template' from integration 'sensor' at packages/ha_lambda_wp_configuration.yaml, line 597: invalid slug EU08L_ambient_operating_state_txt (try eu08l_ambient_operating_state_txt) for dictionary value 'sensors', got {'EU08L_ambient_operating_state_txt': {'friendly_name': 'EU08L_Ambient_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error'][states('sensor.EU08L_ambient_operating_state')|int] }}\n"}, 'EU08L_emgr_operating_state_txt': {'friendly_name': 'EU08L_EMgr_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error', 'Offline'][states('sensor.EU08L_emgr_operating_state')|int] }}\n"}, 'EU08L_hp1_error_state_txt': {'friendly_name': 'EU08L_Hp1_Error_St...

Can I ignore them currently till EU08L will be connected in my network or is there anything which I can adjust in advance?

**Comment by RalfWinter** [comment]
Yes, you can ignore those error messages, they will go away when values for your Lambda are coming in.

**Comment by bs0001** [comment]
Lambda is now connected. I put the IP into the configuration but get the following message at line 597:
Logger: homeassistant.config
Quelle: config.py:1300
Erstmals aufgetreten: 18. September 2024 um 12:36:47 (1 Vorkommnisse)
Zuletzt protokolliert: 18. September 2024 um 12:36:47

Invalid config for 'sensor' from integration 'template' at packages/ha_lambda_wp_configuration.yaml, line 597: invalid slug EU08L_ambient_operating_state_txt (try eu08l_ambient_operating_state_txt) for dictionary value 'sensors', got {'EU08L_ambient_operating_state_txt': {'friendly_name': 'EU08L_Ambient_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error'][states('sensor.EU08L_ambient_operating_state')|int] }}\n"}, 'EU08L_emgr_operating_state_txt': {'friendly_name': 'EU08L_EMgr_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error', 'Offline'][states('sensor.EU08L_emgr_operating_state')|int] }}\n"}, 'EU08L_hp1_error_state_txt': {'friendly_name': 'EU08L_Hp1_Error_St..., please check the docs at https://www.home-assistant.io/integrations/template

![image](https://github.com/user-attachments/assets/984e3291-63c0-4531-b7af-6cc18386924d)

Please do not wonder. I changed all EU13L to EU08L.
But what's still wrong?

**Comment by thecem** [comment]
pls use code formatting: <>.

I assume you use multiple times `sensor:` 
You could only have one same domain as main config.

Pls check if you have to add these config to your main sensor domain.

**Comment by bs0001** [comment]
Sorry, I didn't get you completely. I assume that I have on different places the sensor.
Hopefully you can give me a short info what I can do that it works:

**configuration.yaml**
`# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

homeassistant:
  packages: !include_dir_named packages

powercalc:

template:

  - sensor:
      - name: "Power Total Heizung"
        unique_id: power_total_heating
        state: >
          {{ states('shellypro3em_XXXXXXXX_phase_a_active_power')|float(0) +
             states('shellypro3em_XXXXXXXX_phase_b_active_power')|float(0) +
             states('shellypro3em_XXXXXXXX_phase_c_active_power')|float(0) }}
        unit_of_measurement: W
        device_class: power
        state_class: measurement
        availability: >
            {{
               [ states('shellypro3em_XXXXXXXX_phase_a_active_power'),
                 states('shellypro3em_XXXXXXXX_phase_b_active_power'),
                 states('shellypro3em_XXXXXXXX_phase_c_active_power') ] | map('is_number') | min }}

  - sensor:
      - name: "Energy Total Heizung"
        unique_id: energy_total_heating
        state: >
          {{ states('shellypro3em_XXXXXXXX_total_active_energy')|float(0) }}
        unit_of_measurement: kWh
        device_class: energy
        state_class: total_increasing
        attributes:
          last_reset: "1970-01-01T00:00:00+00:00"
        availability: >
            {{ 
               [ states('shellypro3em_XXXXXXXX_total_active_energy') ] | map('is_number') | min }}

#template: !include template.yaml
template: !include packages/ha_lambda_wp_configuration.yaml`


In subdirectory packages I created the ha_lambda_wp_configuration.yaml with all content of the file "HA-Lambda-WP_configuration.yaml".
I replaced only EU13L to EU08L and put in the correct IP address.

I receive the following error:
Konfigurationswarnungen
Invalid config for 'template' from integration 'sensor' at packages/ha_lambda_wp_configuration.yaml, line 597: invalid slug EU08L_ambient_operating_state_txt (try eu08l_ambient_operating_state_txt) for dictionary value 'sensors', got {'EU08L_ambient_operating_state_txt': {'friendly_name': 'EU08L_Ambient_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error'][states('sensor.EU08L_ambient_operating_state')|int] }}\n"}, 'EU08L_emgr_operating_state_txt': {'friendly_name': 'EU08L_EMgr_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error', 'Offline'][states('sensor.EU08L_emgr_operating_state')|int] }}\n"}, 'EU08L_hp1_error_state_txt': {'friendly_name': 'EU08L_Hp1_Error_St...

My goal is to have an good overview in configuration.yaml and to move all topics which can be moved out to separate yaml files.

BR

**Comment by thecem** [comment]
```
homeassistant:
  packages: !include_dir_named packages
```
and
```

template: !include packages/ha_lambda_wp_configuration.yaml

```
does not work.

**Comment by bs0001** [comment]
yes, for sure. The following line is not correct because alls my packages which are stored in packages folder will automatically read with the following line:
`packages: !include_dir_named packages`

But that solves not my problem. When I delete the line "template: !include packages/ha_lambda_wp_configuration.yaml" I still have the problem.

For an initial test I can also put the whole content of "HA-Lambda-WP_configuration.yaml" at the end of my "configration.yaml". But that also won't work due to the sensor topic.

Therefore it would be helpful to know which parts of the "HA-Lambda-WP_configuration.yaml" should be placed where.

**Comment by thecem** [comment]
Sure this is the reason, you asking for:

you do a:
```

template: 
  sensor:
    - platform: template
      sensors:
        eu13l_ambient_operating_state_txt:
          friendly_name: "EU13L_Ambient_Operating_State_txt"
          value_template: >
            {{ ['Off', 'Automatik', 'Manual', 'Error'][states('sensor.eu13l_ambient_operating_state')|int] }}
```
As I and your log said, this is not working:

`Invalid config for 'sensor' from integration 'template' at packages/ha_lambda_wp_configuration.yaml, line 597:`

**Comment by bs0001** [comment]
and what should I do that it's working?

**Comment by thecem** [comment]
ok, try to hash this line: `template: !include packages/ha_lambda_wp_configuration.yaml`

may you post here your config: ha_lambda_wp_configuration.yaml

I think with packages it should work, i do the same but i use a diff method to map the number to named values.

**Comment by bs0001** [comment]
I already hashed the line.
ha_lambda_wp_configuration.yaml is stored in subdirectory "packages".

Here's my file (I added the .txt at the end, otherwise it was not possible to upload here)

I copied the content of the "HA-Lambda-WP_configuration.yaml", changed the IP address and replaced EU13L to EU08L.

[ha_lambda_wp_configuration.yaml.txt](https://github.com/user-attachments/files/17061702/ha_lambda_wp_configuration.yaml.txt)

**Comment by RalfWinter** [comment]
By the best of my knowledge my Lambda yaml needs to go into the configuration.yaml

**Comment by bs0001** [comment]
Also when I put the whole content of the "HA-Lambda-WP_configuration.yaml" at the end of my configuration.yaml I have the problem with the sensor topic.

This is my configuration.yaml:

```
# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml

homeassistant:
  packages: !include_dir_named packages

powercalc:

template:

  - sensor:
      - name: "Power Total Heizung"
        unique_id: power_total_heating
        state: >
          {{ states('shellypro3em_XXXXXXXX_phase_a_active_power')|float(0) +
             states('shellypro3em_XXXXXXXX_phase_b_active_power')|float(0) +
             states('shellypro3em_XXXXXXXX_phase_c_active_power')|float(0) }}
        unit_of_measurement: W
        device_class: power
        state_class: measurement
        availability: >
            {{
               [ states('shellypro3em_XXXXXXXX_phase_a_active_power'),
                 states('shellypro3em_XXXXXXXX_phase_b_active_power'),
                 states('shellypro3em_XXXXXXXX_phase_c_active_power') ] | map('is_number') | min }}

  - sensor:
      - name: "Energy Total Heizung"
        unique_id: energy_total_heating
        state: >
          {{ states('shellypro3em_XXXXXXXX_total_active_energy')|float(0) }}
        unit_of_measurement: kWh
        device_class: energy
        state_class: total_increasing
        attributes:
          last_reset: "1970-01-01T00:00:00+00:00"
        availability: >
            {{ 
               [ states('shellypro3em_XXXXXXXX_total_active_energy') ] | map('is_number') | min }}
```

It won't work if I copy&paste all of the content of ""HA-Lambda-WP_configuration.yaml" at the end of my file.
May be I have to mix something up with the existing sensor entries. But I really don't know how.

**Comment by bs0001** [comment]
I did it like you mentioned.
I deleted the lines in "packages/ha_lambda_wp_configuration.yaml".

My configuration.yaml looks like now:
![image](https://github.com/user-attachments/assets/eaf1f00e-6af7-41d7-a150-cbb617f3aef3)

When I check my configuration I get the following message:
Konfigurationswarnungen
Invalid config for 'template' from integration 'sensor' at configuration.yaml, line 55: invalid slug EU08L_ambient_operating_state_txt (try eu08l_ambient_operating_state_txt) for dictionary value 'sensors', got {'EU08L_ambient_operating_state_txt': {'friendly_name': 'EU08L_Ambient_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error'][states('sensor.EU08L_ambient_operating_state')|int] }}\n"}, 'EU08L_emgr_operating_state_txt': {'friendly_name': 'EU08L_EMgr_Operating_State_txt', 'value_template': "{{ ['Off', 'Automatik', 'Manual', 'Error', 'Offline'][states('sensor.EU08L_emgr_operating_state')|int] }}\n"}, 'EU08L_hp1_error_state_txt': {'friendly_name': 'EU08L_Hp1_Error_St...

**Comment by thecem** [comment]
Hmm, could you post your complete configuration.yaml without any secure data.

**Comment by bs0001** [comment]
[configuration.yaml.txt](https://github.com/user-attachments/files/17062430/configuration.yaml.txt)

**Comment by thecem** [comment]
For sensors you have to use lower cases, de-hash in the packages yaml and remove it from your yaml. Then try again. If this will be positive @RalfWinter: 
```
homeassistant:
  packages: !include_dir_named packages
```
Is the option to use your file as pack.

**Comment by RalfWinter** [comment]
Thanks mate, I will try this coming weekend.

**Comment by bs0001** [comment]
omg. I changed it to lower caser and no warning came up when I check my configuration.
Hereby the 2 files. Probably useful for storing in the prohect here.

After a HA restart I unfortunately see no eu08l device in my devices. What is the reason therefore?
I assume that I can see the devices and can activate my wished entities,

[configuration.yaml_lowercase.txt](https://github.com/user-attachments/files/17063360/configuration.yaml_lowercase.txt)
[ha_lambda_wp_configuration.yaml_lowercase.txt](https://github.com/user-attachments/files/17063361/ha_lambda_wp_configuration.yaml_lowercase.txt)

**Comment by thecem** [comment]
You don’t have a device since this ist not possible outside mqtt up to now, you have entity’s!

go for developer -> states search for eu08l

**Comment by bs0001** [comment]
ahh, cool, thanks a lot!
I found them under developer --> states as well as in settings --> Devices and services --> Entities

With that info I could write the data into my RasperryMatic system variables and can vizualize them in Homematic. Perfect!
The topic about changings come next. First the vizualization.

@RalfWinter 
Probably some more words in the description might be useful for others.

**Comment by RalfWinter** [comment]
Well, yes, some knowkedge of HA is required ;-)

**Comment by RalfWinter** [comment]
Hi @Briese-henne this is a bug in the modbus implementation of Lambda. The initial version I had on my Lambda did show values, they broke it meanwhile. 
I opened a ticket but nothing happened so far.

**Comment by bs0001** [comment]
> Hi, my HA is writting to the energy manager of the heatpump. Its working fine. Thank you! But some temperature sensors show -300 as value. Can you please help me? Best regards

What did you use on HA to write to lambda heatpump?

**Comment by Briese-henne** [comment]
I am reading from a Cerbo GX from Victron. Im sending negative Values, if there is solar energy  to use.

**Comment by Briese-henne** [comment]
> Hi @Briese-henne this is a bug in the modbus implementation of Lambda. The initial version I had on my Lambda did show values, they broke it meanwhile. I opened a ticket but nothing happened so far.

Thank you, I will wait till they fix it.

**Comment by thecem** [comment]
> > Hi, my HA is writting to the energy manager of the heatpump. Its working fine. Thank you! But some temperature sensors show -300 as value. Can you please help me? Best regards
> 
> What did you use on HA to write to lambda heatpump?

https://github.com/RalfWinter/lambda-heatpump-modbus-tcp-HA/issues/1

**Comment by Briese-henne** [comment]
modbus tcp

**Comment by RalfWinter** [comment]
I guess @thecem meant, whether you used an automation in HA (that is what I use to tell the energy manager my PV surplus).

**Comment by Pitri82** [comment]
> Hi, my HA is writting to the energy manager of the heatpump. Its working fine. Thank you! But some temperature sensors show -300 as value. Can you please help me? Best regards

Hi,
I although have some entities showing -300. After a closer look and a chat with my installation partner, we found out that my lambda is sending -300 for entities where no sensor is installed. For example, I do not have return flow temperature sensors for all heat circuits. So my EU20L can’t send this value. So check if there is a sensor installed on your system.

**Comment by RalfWinter** [comment]
Well, yes, of course, where no sensor there is no data. However, with my initial firmware back in 2022 I saw all set temperatures, since recent firmware updates they are no longer provided via modbus:
![Screenshot_20240921_105858_Home Assistant~2](https://github.com/user-attachments/assets/8938d745-6991-4f22-b9b8-63c29466c3b8)

**Comment by Eisbaer2** [comment]
Die Lambda wird doch eh nur in D & AT verkauft, dann können wir doch auch gleich auf deutsch schreiben, oder nicht?

**Comment by Eisbaer2** [comment]
> > Hi @Briese-henne this is a bug in the modbus implementation of Lambda. The initial version I had on my Lambda did show values, they broke it meanwhile. I opened a ticket but nothing happened so far.
> 
> Thank you, I will wait till they fix it.

Mir wurde vom Lamdba zugesichert, dass sie die Soll-Temperaturen wieder zur Verfügung stellen werden, nur den Zeitpunkt wollten sie mir nicht nennen. Ist schon fast ein Jahr her, und ich bin immer noch bei der 0.0.1 vom 20.9.2023

**Comment by RalfWinter** [comment]
> Die Lambda wird doch eh nur in D & AT verkauft, dann können wir doch auch gleich auf deutsch schreiben, oder nicht?

Da wäre ich mir nicht so sicher, auf der Lambda-Webseite zur Partnersuche (https://lambda-wp.at/partnerfinder/) sehe ich 2 Nadeln in Italien 😉 Ich wäre dafür bei Englisch zu bleiben.

**Comment by Briese-henne** [comment]
Yes, it is an automation.
Unfortunately Lambda modbus tcp is not accepting the writing of negativ values.

the magic happens here:

name: "Victron Total Power Adjusted"
        unit_of_measurement: "W"
        state: >
          {% set value = states('sensor.victron_total_power') | int %}
          {{ value + 65536 if value < 0 else value }}
        device_class: power
The send value is now always positiv . The heatpump GUI is showing it as negativ value.

**Comment by bs0001** [comment]
In the lambda Konfiguration E-Manager you can define the E-Meter Messpunkt.
You can define "Neg.  E-Überschuss" or "Pos. E-Überschuss".
Probably this helpf to work with both variants.

**Comment by thecem** [comment]
I recommend to use this: 
https://github.com/RalfWinter/lambda-heatpump-modbus-tcp-HA/issues/1#issuecomment-1549906446

It is positive and negative failsave, even for a victron sensor.

Define: „Neg. E-Überschuss“

**Comment by RalfWinter** [comment]
To send PV-surplus to my Lambda heatpump I did following:
1. defined "Neg. EÜberschuss" in the E-Manager configuration (this needs higher rights than enduser, I contacted Sigmatek in Germany to set this (if not already done the E-Manager needs activation too)
2.  Added following automation:
```
alias: WP-PV-Excess
description: Sent PV-Excess to HP
trigger:
  - platform: time_pattern
    seconds: /30
condition: []
action:
  - service: modbus.write_register
    data:
      hub: EU13L
      address: 102
      value:
        - >-
          {{states('sensor.shelly3em_total_power_power_total') | int |
          bitwise_and(65535)}}
mode: single
```

The sensor shelly3em_total_power_power_total has negative values when returning power to grid and positive values when geeting power from grid. So, if your power meter operates the same, you can use the automation and replace "sensor.shelly3em_total_power_power_total" with the entity in your system which provides the same type of information (+ / - watts to/from your your premises).

In case you want to use a Shelly 3EM as I did, you need to add a template sensor in your configuration.yaml in order to get above entity "shelly3em_total_power_power_total" which is the combined power over all 3 phases:

```
template:
- sensor:
    - name: "Shelly3EM Total Power"
      unique_id: shelly3em_power_total
      unit_of_measurement: "W"
      state: >
         {{
         states('sensor.shelly_em3_channel_a_power')| float(0) +
         states('sensor.shelly_em3_channel_b_power')| float(0) +
         states('sensor.shelly_em3_channel_c_power')| float(0) | round(0)
         }}

```

Remark: the advantage with Shelly devices is that they natively integrate into home assistant. Besides the 3 phase meter I use a number of switches and dimmer.

**Comment by thecem** [comment]
be aware that U have in:

```
alias: WP-PV-Excess
description: Sent PV-Excess to HP
trigger:
  - platform: time_pattern
    seconds: /30
condition: []
action:
  - service: modbus.write_register
    data:
      hub: EU13L
      address: 102
      value:
        - >-
          {{states('sensor.power_current') | int |
          bitwise_and(65535)}}
mode: single 
```

the "-" in front of  ">-" the template since HA is only calling modbus function code 0x10 if you define an array!

**Comment by RalfWinter** [comment]
Thanks @thecem . Was your comment directed to me or any one else you wants to implement the automation?

**Comment by prodigy9999** [comment]
@RalfWinter Do you have the Lambda-WP_configuration.yaml also in german language? Perhaps you could share your files?
At the end I would like to set the temperature in HA with a slider via Modbus write. any idea for it?

**Comment by RalfWinter** [comment]
Hi Christian, yes, I use a German version for the entity names (the values themselves are in English). I can upload it on the weekend.

I can't help on the slider question.

**Comment by thecem** [comment]
Hier ist eine Home Assistant Automatisierung, die einen Input Slider verwendet, um einen Wert in einem Modbus Hub mit dem Namen EU08L im Register 1234 zu setzen und diesen Wert ebenfalls auszulesen:

1. Konfiguration des Modbus Hubs

Zunächst muss der Modbus Hub und das Register in der configuration.yaml definiert werden. Der Hub EU08L muss entsprechend konfiguriert sein:

```
modbus:
  - name: "EU08L"
    type: tcp
    host: 192.168.1.100  # Ersetze durch die IP-Adresse des Modbus-Geräts
    port: 502

    sensors:
      - name: "Temperatur"
        slave: 1
        address: 1234
        input_type: holding
        unit_of_measurement: "°C"
        scan_interval: 10  # Zeitintervall in Sekunden zum Auslesen des Wertes

    registers:
      - name: "Set_Temperatur"
        slave: 1
        address: 1234
        register_type: holding
        count: 1
```

2. Erstellung des Input Sliders

Erstelle einen Input Slider, um den Temperaturwert einzustellen:

```
input_number:
  temperatur:
    name: Temperatur
    initial: 20
    min: 15
    max: 30
    step: 0.5
    unit_of_measurement: "°C"
```

3. Automatisierung zum Setzen und Auslesen des Wertes

Erstelle eine Automatisierung, die jedes Mal ausgelöst wird, wenn der Slider geändert wird. Hierbei wird der Wert im Modbus Register gesetzt und anschließend ausgelesen.

Automatisierung für das Setzen des Wertes:

```
automation:
  - alias: "Setze Temperatur im Modbus"
    trigger:
      platform: state
      entity_id: input_number.temperatur
    action:
      - service: modbus.write_register
        data:
          hub: EU08L
          unit: 1
          address: 1234
          value: "{{ states('input_number.temperatur') | int }}"
```

Automatisierung für das Auslesen des Wertes:

```
automation:
  - alias: "Lese Temperatur vom Modbus aus"
    trigger:
      platform: time_pattern
      minutes: "/1"  # Alle 1 Minute den Wert auslesen
    action:
      - service: input_number.set_value
        data:
          entity_id: input_number.temperatur
          value: "{{ states('sensor.temperatur') | float }}"
```

4. Zusammenfassung:

	•	Der Input Slider wird verwendet, um den Wert zu setzen.
	•	Sobald der Slider geändert wird, wird der neue Wert in das Modbus Register 1234 geschrieben.
	•	Ein separates Trigger-Event liest den Wert des Registers periodisch aus und aktualisiert den Slider entsprechend.

Diese Konfiguration sollte es ermöglichen, den Slider zur Steuerung der Temperatur zu verwenden und gleichzeitig den aktuellen Wert im Modbus Register anzuzeigen.

**Comment by RalfWinter** [comment]
> @RalfWinter Do you have the Lambda-WP_configuration.yaml also in german language? Perhaps you could share your files? At the end I would like to set the temperature in HA with a slider via Modbus write. any idea for it?

I just looked into your request: personally I only use English language, for entities and values. There was another request for German language, however, I already denied because Lambda is not limited to the German and Austrian market. Therefore I keep this in English for now.

If there would be an easy way to maintain multiple languages in YAML depending on which language somebody selected in the HA-GUI, then it would make sense to add other languages, e.g. German.
